### ISSUE TITLE
Error when link #access property is a boolean

### ISSUE THREAD

**Issue by claudiu-cristea** [openeuropa/oe_bootstrap_theme#285]
## Problem

In `MenuPreprocess::prepareLocalTasks()` a check to link `#access` assumes that this property is always an instance of `AccessResultInstance`. But this isn't always the case, link elements, as all render/form elements, accept also a _boolean_ as value for `#access`. Some core examples:

* https://git.drupalcode.org/project/drupal/-/blob/9.5.x/core/modules/layout_builder/src/Form/LayoutBuilderEntityViewDisplayForm.php#L67
* https://git.drupalcode.org/project/drupal/-/blob/9.5.x/core/modules/layout_builder/src/Element/LayoutBuilder.php#L374
* https://git.drupalcode.org/project/drupal/-/blob/9.5.x/core/modules/layout_builder/src/Controller/ChooseBlockController.php#L122
* https://git.drupalcode.org/project/drupal/-/blob/9.5.x/core/modules/comment/src/Form/CommentAdminOverview.php#L218
* ...and so on

When a link having `#access` set as boolean is processed in that method, this error is triggered:

```
Error: Call to a member function isAllowed() on bool in Drupal\oe_bootstrap_theme\MenuPreprocess->prepareLocalTasks() (line 196 of themes/contrib/oe_bootstrap_theme/src/MenuPreprocess.php).
```

## Proposal

Accept also _booleans_ as values for `#access`.

**Comment by claudiu-cristea** [comment]
I solve this issue in our project by detecting the code where `#access` is set to a boolean and replacing with it with an `AcessResultInterface` instance. That's why I will not continue to fix #286, that still needs test coverage. However, I think this issue is still relevant and should be fixed.

**Comment by drishu** [comment]
About moving the access check out of the preprocess and letting the renderer take care of it, the BCL repo is not prepared for render arrays but our approach will change in the next major version.
